### PR TITLE
Print value of invalid mode

### PR DESCRIPTION
**Pull request by 5yler**
### Background

- Print integer value of invalid mode for debugging purposes

### What's new
- ✅ [New feature description. Only a single new major feature is allowed per PR.]
- ❌ New feature is accompanied by new test: [name and description of new test].

### Tests
1. ✅Tested on simulated robot: [Describe the tests/commands used.]
2. ❌ Tested on physical robot: [Describe the tests/commands used.]

### Quality
3. ✅ New code is written in pure C++17 to the best of my knowledge.
4. ✅ New code follows established C++17 best practices ([C++ Core Guidelines](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines)).
5. ✅New code passed `clang-format` and `cpplint` at least, if not all of our static code analysers.
6. ❌ I have included Doxygen-style documentation in the new C++ code.

### Note to reviewers
Please verify that all sections are accurately filled out and that all 6 numbered entries above are present and ticked/checked off, with their requirements met, if applicable.

### CODE REVIEW

File: src/driver/communication_interface.cc [hunk 1]
```diff
@@ -387,6 +387,7 @@ bool CommunicationInterface::ModeIsValid(
     case franka::RobotMode::kAutomaticErrorRecovery:
       return true;
   }
+  log()->error("ModeIsValid: got invalid mode: {}", current_mode);
```
Comment (duanyutong) [inline src/driver/communication_interface.cc:390]: need to convert `current_mode` to string with `RobotModeToString`

Comment (5yler) [reply src/driver/communication_interface.cc:390]: No, that won't work, because it is not a valid mode. It does not exist in the enum of known modes.

Comment (5yler) [reply src/driver/communication_interface.cc:390]: I want the raw integer value

**Review by duanyutong — APPROVED** [review]

File: src/driver/communication_interface.cc (hunk 1 above)
Comment (duanyutong) [reply src/driver/communication_interface.cc:390]: Yep you are right. Do you need a cast here?

**Pull request merged**